src/hooks/useSwapSlippageTolerance.ts

```diff
@@ -11,31 +11,27 @@ import useGasPrice from './useGasPrice'
 import useUSDCPrice, { useUSDCValue } from './useUSDCPrice'
 import { useActiveWeb3React } from './web3'
 
-const V2_SWAP_DEFAULT_SLIPPAGE = new Percent(50, 10_000) // .50%
 const V3_SWAP_DEFAULT_SLIPPAGE = new Percent(50, 10_000) // .50%
 const ONE_TENTHS_PERCENT = new Percent(10, 10_000) // .10%
 
 /**
  * Return a guess of the gas cost used in computing slippage tolerance for a given trade
  * @param trade the trade for which to _guess_ the amount of gas it would cost to execute
  */
-function guesstimateGas(
-  trade: V2Trade<Currency, Currency, TradeType> | V3Trade<Currency, Currency, TradeType> | undefined
-): number | undefined {
+function guesstimateGas(trade: Trade<Currency, Currency, TradeType> | undefined): number | undefined {
   if (trade instanceof V2Trade) {
```

Comment:
@NoahZinsmeister how would you reconcile this here? We only have a router-sdk/Trade object to wor kwith. Should we stop estimating gas for v2?

Reply:
Related - teh auto router only returns gas cost estimate for V3 currently right?

Reply:
@NoahZinsmeister so basically always run the existing logic we had for v3

@ianlapham possibly yeah in your PR where you carry the gas estimate around we could use it. We do however still fall back to the legacy v3 router some times so we need a solution for that case.